config/configtls/configtls_test.go

```diff
@@ -201,3 +206,166 @@ func TestLoadTLSServerConfig(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, tlsCfg)
 }
+
+func TestCertificateReload(t *testing.T) {
+	t.Run("Should eagerly load the given certificate", func(t *testing.T) {
```

Comment:
Consider to have 3 tests instead of a test with 3 `t.Run`

Reply:
I'll refactor this into something that looks more like a table test.